src/io/iohk/scalanet/peergroup/PeerGroup.scala

```diff
@@ -12,25 +12,27 @@ sealed trait PeerGroup[A] {
   val processAddress: A
   def initialize(): Task[Unit]
   def sendMessage[MessageType: Codec](address: A, message: MessageType): Task[Unit]
-  def messageChannel[MessageType: Codec]: Observable[MessageType]
+  def messageChannel[MessageType: Codec]: Observable[(A, MessageType)]
```

Comment:
As per the other comments from Alexis, it's a good idea to group this things into a case class instead of using tuples